src/backend/distributed/connection/placement_connection.c

```diff
@@ -559,6 +559,8 @@ StartColocatedPlacementConnection(uint32 flags, ShardPlacement *placement,
 	 */
 	if (placement->partitionMethod != DISTRIBUTE_BY_HASH)
 	{
+		// TODO: This is where we place a function mapping groupid -> nodename, nodeport
+		// flags can later tell us whether secondaries are okay
```

Comment:
@marcocitus My current plan is to remove nodename and nodeport from `ShardPlacement`. That would mean that here we lookup which node to actually connect to for a given group. Do you think that's reasonable considering a future where we might want to add logic which returns the preferred node given some information on the current query/transaction?